src/actions/sso/consumeInteractionToken.ts

```diff
@@ -0,0 +1,62 @@
+import { JolocomLib } from 'jolocom-lib'
+import { interactionHandlers } from '../../lib/storage/interactionTokens'
+import { withErrorScreen, withLoading } from '../modifiers'
+import { showErrorScreen } from '../generic'
+import { AppError } from '../../lib/errors'
+import ErrorCode from '../../lib/errorCodes'
+import { ThunkAction } from '../../store'
+import {
+  JSONWebToken,
+  JWTEncodable,
+} from 'jolocom-lib/js/interactionTokens/JSONWebToken'
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const callHandler = (handler: () => ThunkAction) => {
+  try {
+    return handler()
+  } catch (e) {
+    if (e.message === 'Signature on token is invalid') {
```

Comment:
I would like to fix these checks to use proper error codes before this code is merged!

Reply:
These have been added